### PR TITLE
apache-spark 3.3.0

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by Porkepix** [comment]
I was looking at it earlier, I believe `openjdk` can be upgraded to at least 17 (latest, 18 might be fine, but their website only mention LTS ones, `8`, `11` and `17`).

**Comment by chenrui333** [comment]
Agree, I saw 17 [is part of build](https://github.com/apache/spark/blob/master/.github/workflows/build_java17.yml#L20)

**Comment by chenrui333** [comment]
> Spark runs on Java 8/11/17, Scala 2.12/2.13, Python 3.7+ and R 3.5+. Java 8 prior to version 8u201 support is deprecated as of Spark 3.2.0. For the Scala API, Spark 3.3.0 uses Scala 2.12. You will need to use a compatible Scala version (2.12.x).

yeah, let's use java 17

File: Formula/apache-spark.rb [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
+  sha256 "306b550f42ce1b06772d6084c545ef8448414f2bf451e0b1175405488f2a322f"
   license "Apache-2.0"
   head "https://github.com/apache/spark.git", branch: "master"
 
   bottle do
     sha256 cellar: :any_skip_relocation, all: "3f8f3309dfef579496100aad618c4f46b55795497fe259f2accb0da0c15da7ec"
   end
 
-  depends_on "openjdk@11"
+  depends_on "openjdk"
```
Comment (carlocab) [inline Formula/apache-spark.rb:15]: Note that this is Java 18, not Java 17.

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**